dask_match/expr.py

```diff
@@ -887,6 +907,81 @@ def simplify(self):
                 for op in self.frame.operands
             ]
             return type(self.frame)(*operands)
+        elif isinstance(self.frame, PartitionsFiltered):
+            if self.frame._partitions:
+                partitions = [self.frame._partitions[p] for p in self.partitions]
+            else:
+                partitions = self.partitions
+            # We assume that expressions defining a special "_partitions"
+            # parameter can internally capture the same logic as `Partitions`
+            operands = [
+                partitions if self.frame._parameters[i] == "_partitions" else op
+                for i, op in enumerate(self.frame.operands)
+            ]
+            return type(self.frame)(*operands)
+
+
+class PartitionsFiltered(Expr):
+    """Mixin class for partition filtering
+
+    A `PartitionsFiltered` subclass must define a
+    `_partitions` parameter. When `_partitions` is
+    defined, the following expresssions must produce
+    the same output for `cls: PartitionsFiltered`:
+      - `cls(expr: Expr, ..., _partitions)`
+      - `Partitions(cls(expr: Expr, ...), _partitions)`
+
+    In order to leverage the default `Expr._layer`
+    method, subclasses should define `_filtered_task`
+    instead of `_task`.
+    """
+
+    @property
+    def _filtered(self) -> bool:
+        """Whether or not output partitions have been filtered"""
+        return (
+            "_partitions" in self._parameters
```

Comment:
I'm imagining that `"_partitions" in self._parameters` is as an expectation of this class.